include/sparrow/types/arrow_length.hpp

```diff
@@ -0,0 +1,166 @@
+                        std::string("size/length/offset cannot be represented by ") + typeid(TargetRepr).name()
+                        + ": " + std::to_string(size_or_offset) + " (" + typeid(size_or_offset).name() + ")"
+                    );
+                }
+            }
+        }
+    }
+
+
+#if defined(__GNUC__) && !defined(__clang__)
+#   pragma GCC diagnostic push
+#   pragma GCC diagnostic ignored "-Wuseless-cast" // We want to be able to cast type aliases to their real types.
+#endif
+
+    /// @returns The provided arrow length value as represented by the native standard size type `std::size_t`.
+    ///          If `config::enable_size_limit_runtime_check == true` it will also check that the value is
+    ///          a valid arrow length and representable by `std::size_t` or throws otherwise.
+    /// @throws  @see `throw_if_invalid_size()` for details.
+    inline constexpr
+    auto to_native_size(arrow_length length) -> std::size_t
+    {
+        throw_if_invalid_size<std::size_t>(length);
+        return static_cast<std::size_t>(length);
+    }
+
+    /// @returns The provided arrow length value as represented by the native standard offset type `std::ptrdiff_t`.
+    ///          If `config::enable_size_limit_runtime_check == true` it will also check that the value is
+    ///          a valid arrow length and representable by `std::ptrdiff_t` or throws otherwise.
+    /// @throws  @see `throw_if_invalid_size()` for details.
+    inline constexpr
+    auto to_native_offset(arrow_length offset) -> std::ptrdiff_t
+    {
+        throw_if_invalid_size<std::ptrdiff_t>(offset, arrow_length_kind::offset);
+        return static_cast<std::ptrdiff_t>(offset);
+    }
+
+    /// @returns The provided size or offset value as represented by an arrow-length type (int64_t).
+    ///          If `config::enable_size_limit_runtime_check == true` it will also check that the value is
+    ///          a valid arrow length and representable in `int64_t` or throws otherwise.
+    /// @throws  @see `throw_if_invalid_size()` for details.
+    inline constexpr
+    auto
+    to_arrow_length(std::integral auto size_or_offset, arrow_length_kind kind = arrow_length_kind::size)
+        -> arrow_length
+    {
+        throw_if_invalid_size(size_or_offset, kind);
+        return static_cast<arrow_length>(size_or_offset);
+    }
+
+    /// @returns The sum of the provided offsets with `R` representation, whatever the offset types as long as
+    ///          they are integrals and can represent and arrow length.
+    ///          If `config::enable_size_limit_runtime_check == true` it will also check that each value and the resulting sum are
+    ///          valid arrow lengths and representable in the specified result `R` or throws otherwise.
+    /// @throws  @see `throw_if_invalid_size()` for details.
+    template<std::integral R = arrow_length>
+    inline constexpr
+    auto sum_arrow_offsets(std::integral auto... offsets) -> R
```

Comment:
We have some calls like [here](https://github.com/man-group/sparrow/blob/2baa1ed03f9cc8a0018e9e0f06307ad6ea4bda46/include/sparrow/buffer/buffer_adaptor.hpp#L481)

Reply:
? I dont understand  your comment at all...

my question is, why do we need a variadic function for that

Reply:
Yeah forget about the example, it is unrelated. I'll replace it with a binary function.

Reply:
It's best to check the usage of these functions in [the previous PR](https://github.com/man-group/sparrow/pull/179/files#diff-6d1e508525bb97f11d4a493850e0fe5497c77c8269d4e28be88a254e069c3158L341) they are coming from to better understand why I ended up adding it.
Another example: https://github.com/man-group/sparrow/pull/179/files#diff-750d798de01fd1871b3e5e355d4c9b238d05a97a2e9a6975a40ca8b725b9bed5R157

You can find many examples in there.

While this function is not mandatory, it's utility is to:
- make sure that all the argument of such sum are in fact valid arrow length (which was not always the case in the previous pr, so that helped fixed that);
- when runtime checks are enabled, this is a check point to make sure the computed size is indeed in the required range (64bit or 32bit if the user want to constrain to it).

> But probably nowhere in the sparrow codebase we would use this, as this implies the length of the offsets (ie the number of variadic arguments) is known at compile time

In the usage from the previous PR you can see that we are always using 2 arguments with this functions, so it could be a 2 argument function, if you prefer. That wouldnt be specially superior to such version that takes as many argument as you need, but given the usage that wouldnt change a thing.